src/librustc/ty/context.rs

```diff
@@ -2221,12 +2221,12 @@ impl<'a, 'gcx, 'tcx> TyCtxt<'a, 'gcx, 'tcx> {
 
     pub fn mk_param(self,
                     index: u32,
-                    name: Name) -> Ty<'tcx> {
+                    name: InternedString) -> Ty<'tcx> {
```

Comment:
this change seems fine to me (as discussed on IRC); name should only be used for debug print-outs etc